src/components/__tests__/AutoComplete.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { shallow } from "enzyme";
+import { createElement } from "react";
+
+import * as Autosuggest from "react-autosuggest";
+
+import { AutoComplete, AutoCompleteProps } from "../AutoComplete";
+
+describe("AutoComplete", () => {
+    const renderAutoComplete = (props: AutoCompleteProps) => shallow(createElement(AutoComplete, props));
+    const defaultProps: AutoCompleteProps = {
+        addTag: () => { /* */ },
+        inputPlaceholder: "",
+        lazyLoad: false,
+        suggestions: []
+    };
+
+    it("renders the structure correctly", () => {
+        const autoComplete = renderAutoComplete(defaultProps);
+
+        expect(autoComplete).toBeElement(
+            createElement(Autosuggest, {
+                getSuggestionValue: jasmine.any(Function),
+                inputProps: jasmine.any(Object),
+                onSuggestionSelected: jasmine.any(Function),
+                onSuggestionsClearRequested: jasmine.any(Function),
+                onSuggestionsFetchRequested: jasmine.any(Function),
+                renderSuggestion: jasmine.any(Function),
+                suggestions: []
+            })
+        );
+    });
+
+    it("fetches the suggestions when value is specified", () => {
+        const suggestion = {
+            name: "Uganda",
+            newValue: "Netherlands",
+            suggestionValue: "Uganda",
+            value: "Uganda",
+            method: "enter"
+        };
+        const autoComplete = renderAutoComplete(defaultProps);
+
+        const autoCompleteInstance = autoComplete.instance() as any;
+        autoCompleteInstance.onSuggestionsFetchRequested(suggestion);
```

Comment:
And space

Reply:
Tag instance failed to spy on private functions and using any is the only way to get it working